packages/x-date-pickers/src/internals/hooks/usePicker/usePickerProvider.ts

```diff
@@ -136,3 +147,14 @@ export interface UsePickerProviderProps extends FormProps {
    */
   orientation?: PickerOrientation;
 }
+
+/**
+ * Props used to create the picker's contexts and that are not available on static pickers.
+ */
+export interface UsePickerProviderNonStaticProps {
+  /**
+   * If `true`, the open picker button will not be rendered (renders only the field).
+   * @default false
+   */
+  disableOpenPicker?: boolean;
```

Comment:
After #15671, it's `usePickerProvider` that will generate the info saying if the opening UI should be rendered or not.
So I'm moving the definition of this prop here.

Note that this prop will likely go away in the future, people can use the field component directly.